### PR TITLE
Enhance crm_report

### PR DESCRIPTION
**Pull request by kgaillot**
This is a significant overhaul of crm_report, partly for performance improvements and general cleanup, but primarily to better support Pacemaker Remote nodes.

This addresses the most significant concerns in RHBZ#1323544.

### CODE REVIEW

File: tools/report.collector [hunk 1]
```diff
@@ -220,27 +225,35 @@ getconfig() {
...
 	echo "$host" > $target/STOPPED
     fi
 
     if [ -f "$target/$CIB_F" ]; then
 	crm_verify -V -x $target/$CIB_F >$target/$CRM_VERIFY_F 2>&1
-	CIB_file=$target/$CIB_F crm configure show >$target/$CIB_TXT_F 2>&1
+        if which crm >/dev/null 2>&1 ; then
+            CIB_file=$target/$CIB_F crm configure show >$target/$CIB_TXT_F 2>&1
+        elif which pcs >/dev/null 2>&1 ; then
+            pcs config -f $target/$CIB_F >$target/$CIB_TXT_F 2>&1
```
Comment (beekhof) [inline tools/report.collector]: If only I had a dollar for every time I was looking at an archive and thought "we should really do pcs config here" and never came back :-)

File: tools/report.collector [hunk 2]
```diff
@@ -615,17 +615,21 @@ get_logfiles() {
...
-    findmsg 3 cib_perform_op
-    findmsg 3 process_lrm_event
+    # - make sure we get something from the Policy Engine
+    pattern="$pattern\\|Calculated Transition"
 
+    # - cib and lrmd updates (helpful on non-DC nodes or when the cluster has been up for a long time)
+    pattern="$pattern\\|cib_perform_op\\|process_lrm_event"
+
+    # - pacemaker_remote might use a different file
+    pattern="$pattern\\|pacemaker_remoted:"
```
Comment (beekhof) [inline tools/report.collector]: Excellent idea!

File: tools/report.collector [hunk 3]
```diff
@@ -410,7 +410,7 @@ getbacktraces() {
 
 	# Make a copy of them in case we need more data later
 	# Luckily they compress well
-	mkdir cores &> /dev/null
+	mkdir cores >/dev/null 2>&1
```
Comment (beekhof) [inline tools/report.collector]: But I like bashisms :)

File: tools/report.common.in [hunk 4]
```diff
@@ -336,34 +346,58 @@ linetime() {
...
 }
 
-# Find pattern in a logfile somewhere
-# Return $max ordered results by age (newest first)
+#
+# findmsg <max> <pattern>
+#
+# Print the names of up to <max> system logs that contain <pattern>,
+# ordered by most recently modified.
+#
```
Comment (beekhof) [inline tools/report.common.in]: Like++

File: tools/crm_report.in [hunk 5]
```diff
@@ -365,67 +365,84 @@ do_cts() {
...
-	# Look in the logs...
-	logfile=`findmsg 1 "crm_update_peer"`
-	debug "Reading nodes from $logfile"
-	if [ ! -z "$logfile" ]; then
-	    grep crm_update_peer: $logfile | sed s/.*crm_update_peer// | sed s/://g | awk '{print $2}' | grep -v "(null)" | sort -u | tr '\n' ' '
-	fi
+    # TODO: This has multiple issues:
+    # * It looks for messages from crm_update_peer(), which is used only by
+    #   heartbeat and legacy plugin clusters; it should work with CMAN and
+    #   corosync2 clusters as well.
```
Comment (beekhof) [inline tools/crm_report.in]: Switch to crm_update_peer_state:

**Comment by beekhof** [comment]
Really excellent work

File: tools/report.collector (hunk 1 above)
Comment (kgaillot) [inline tools/report.collector]: That and skipping hb_uuid.txt on non-heartbeat clusters were the most satisfying parts ;)

**Pull request merged**